aws/cluster/elb-dns/ingress.tf

```diff
@@ -12,14 +12,20 @@ data "aws_route53_zone" "current" {
 data "aws_elb_hosted_zone_id" "current" {
 }
 
+# this is a workaround as aws_elb_hosted_zone_id doesn't support NLBs 
+# ref: https://github.com/hashicorp/terraform-provider-aws/issues/7988
+data "aws_lb" "current" {
+  name = split("-", data.kubernetes_service.current.status[0].load_balancer[0].ingress[0].hostname).0
+}
+
 resource "aws_route53_record" "host" {
   zone_id = data.aws_route53_zone.current.zone_id
   name    = var.metadata_fqdn
   type    = "A"
 
   alias {
     name                   = data.kubernetes_service.current.status[0].load_balancer[0].ingress[0].hostname
-    zone_id                = data.aws_elb_hosted_zone_id.current.id
+    zone_id                = var.using_nlb ? data.aws_lb.current.zone_id : data.aws_elb_hosted_zone_id.current.id
```

Comment:
Thanks for the contribution. Just one small issue, the conditional you use for the zone_id here, also needs to be used for the wildcard entry below.

Reply:
Nice catch, updated it